qiskit/quantum_info/operators/operator.py

```diff
@@ -199,6 +199,49 @@ def from_label(cls, label):
                 op = op.compose(label_mats[char], qargs=[qubit])
         return op
 
+    @classmethod
+    def from_circuit(cls, circuit, use_layout=True, layout=None):
+        """Create a new Operator object from a :class`.QuantumCircuit`
+
+        While a :class:`.QuantumCircuit` object can passed directly as ``data``
+        to the class constructor
+
+        By default this constructor method will permute the qubits based on a
+        configured initial layout (i.e. after it was transpiled). It also
+        provides an option to manually provide a :class:`.Layout` object
+        directly.
+
+        Args:
+            circuit (QuantumCircuit): The :class:`.QuantumCircuit` to create an Operator
+                object from.
+            use_layout (bool): When set to ``False`` if the input ``circuit``
+                has a layout set it will be ignored
+            layout (Layout): If specified this kwarg can be used to specify a
+                particular layout to use to permute the qubits in the created
+                :class:`.Operator`. If this is specified it will be used instead
+                of a layout contained in the ``circuit`` input.
```

Comment:
It's slightly unusual to me that `use_layout` is ignored if `layout` is passed, given how similar they are.  I'm not at all married to the idea, but how would it feel to make the type of `layout` a Boolean or a `Layout` - `True` means take from the circuit if available, `False` means ignore all, and `Layout` means use this as an override?

Reply:
I can change it if you'd prefer that pattern. TBH, that seems a bit weirder to have mixed types on a single arg, but there really isn't a good pattern here. Maybe rename `use_layout` to something like `use_set_layout` or `use_bundled_layout` to indicate it means use the `_layout` property that gets set in the circuit by `transpile()`. Or alternatively we can just remove it as a flag because using the specified layout set layout is really the only difference between `Operator(circuit)` and `Operator.from_circuit(circuit)` so instead of having a flag we can just tell people to call `__init__` if that's the behavior they want.

Reply:
No, I think your method is better.  Perhaps flip it, and call it `ignore_set_layout`?  That's maybe cleaner - it's slightly odd that `Operator.from_circuit(qc, layout=..., use_layout=True)` will ignore `use_layout`, but it seems more natural that `Operation.from_circuit(qc, layout=..., ignore_set_layout=True)` won't matter.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7616/commits/b6ebe91668be88faf76dca608a623d46a1770135